springwolf-plugins/springwolf-amqp-plugin/src/main/java/io/github/stavshamir/springwolf/producer/SpringwolfAmqpProducer.java

```diff
@@ -1,28 +1,58 @@
 package io.github.stavshamir.springwolf.producer;
 
+import com.asyncapi.v2.binding.amqp.AMQPChannelBinding;
+import com.asyncapi.v2.binding.amqp.AMQPOperationBinding;
+import com.asyncapi.v2.model.channel.ChannelItem;
+import com.asyncapi.v2.model.channel.operation.Operation;
+import io.github.stavshamir.springwolf.asyncapi.ChannelsService;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.amqp.rabbit.core.RabbitTemplate;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
 
+import java.util.List;
 import java.util.Map;
 
 @Slf4j
 @Service
 public class SpringwolfAmqpProducer {
 
+    private final ChannelsService channelsService;
     private final RabbitTemplate rabbitTemplate;
 
     @Getter
     private boolean isEnabled = true;
 
-    public SpringwolfAmqpProducer(@Autowired RabbitTemplate rabbitTemplate) {
-        this.rabbitTemplate = rabbitTemplate;
+    public SpringwolfAmqpProducer(ChannelsService channelsService, List<RabbitTemplate> rabbitTemplates) {
+        this.channelsService = channelsService;
+        this.rabbitTemplate = rabbitTemplates.get(0);
```

Comment:
The difference between multiple rabbit template beans is preset default exchange and/or routing key. So you could use the overload .send method with just routingKey as param and payload, or just payload.
Because we're getting the exchange name and routing key from the context and we're using the method with all parameters (i.e. exchange, routing key and payload) we don't care which rabbit template to use.

Reply:
Got it, thanks